packages/manager/apps/web/client/app/hosting/database/order/components/private-database/private-database.service.js

```diff
@@ -0,0 +1,55 @@
+import find from 'lodash/find';
+import sortBy from 'lodash/sortBy';
+
+import { CATALOG_PRODUCT } from './private-database.constant';
+
+export default class OrderPrivateDatabaseService {
+  /* @ngInject */
+  constructor($filter, Hosting, WucOrderCartService) {
+    this.$filter = $filter;
+    this.Hosting = Hosting;
+    this.WucOrderCartService = WucOrderCartService;
+  }
+
+  static getCatalogProducts(catalog, products) {
+    if (!products || !products.length) {
+      return [];
+    }
+    const { category } = CATALOG_PRODUCT;
+    return products.map((item) =>
+      find(catalog[category], { planCode: item.planCode }),
+    );
+  }
+
+  async getCatalog(ovhSubsidiary) {
+    const { name } = CATALOG_PRODUCT;
+    return this.WucOrderCartService.getProductPublicCatalog(
+      ovhSubsidiary,
+      name,
+    );
+  }
+
+  async getServices() {
+    const services = await this.Hosting.getHostings();
+    return services.sort();
+  }
+
+  getDatacenter(serviceName) {
+    return this.Hosting.getHosting(serviceName).then(
+      (hosting) => hosting.datacenter,
+    );
+  }
+
+  async getProducts(catalog) {
```

Comment:
I dont see an asynchronous code here. Is async required?
```suggestion
  getProducts(catalog) {
```